pkg/controller/manager/manager_controller.go

```diff
@@ -315,13 +315,12 @@ func (r *ReconcileManager) Reconcile(request reconcile.Request) (reconcile.Resul
 	var internalTrafficSecret *corev1.Secret
 	if management {
 
-		// If clusterType is management and the customer brings its own cert, copy it over to the manager ns.
+		// We expect that the secret that holds the certificates for tunnel certificate generation
+		// is already created by the Api Server
 		tunnelSecret = &corev1.Secret{}
 		err := r.client.Get(ctx, client.ObjectKey{Name: render.VoltronTunnelSecretName, Namespace: render.OperatorNamespace()}, tunnelSecret)
 		if err != nil {
 			if errors.IsNotFound(err) {
-				tunnelSecret = nil
-			} else {
```

Comment:
I don't think there is any need for the `IsNotFound` check. I believe any error for the secret should result in SetDegraded here?